### ISSUE TITLE
template_file provider does not report line number or variable for syntax errors

### ISSUE THREAD

**Issue by ketzacoatl** [hashicorp/terraform#3218]
This makes for a situation that is difficult to debug:

```
...
Error applying plan:

2 error(s) occurred:

* template_file.services-init: failed to render scripts/init-services.tpl: parse error: syntax error
* template_file.compute-init: failed to render scripts/init-compute.tpl: parse error: syntax error
```

What errors and where, can Terraform tell me?

**Comment by zot420** [comment]
+1

It's very difficult to troubleshoot this. I even get it using the stock example from the docs.

**Comment by tyrostone** [comment]
+1

**Comment by jniesen** [comment]
+1

**Comment by jantman** [comment]
I'll vote for this too. I was only able to find the error by applying every git commit, to find the one that introduced the failure.

This problem is made worse by the fact that ``plan`` doesn't seem to actually evaluate the template, so it doesn't show an error. Not only do we not get any further feedback, but the syntax error is only visible on an actual ``apply``.

**Comment by endofcake** [comment]
+1

**Comment by CaptBeardie** [comment]
+1

**Comment by joaocc** [comment]
Any update on this?

**Comment by jonaskint** [comment]
+1

**Comment by apparentlymart** [comment]
It does look like we could run `hil.Parse` in the validate func to get some initial template syntax validation before apply.

For the actual error contents, it looks like Terraform is just passing through verbatim the error returned from `hil.Parse`, so getting a source location here will probably require changes to [`hil`](https://github.com/hashicorp/hil).

**Comment by zutherb** [comment]
+1

**Comment by trawler** [comment]
👍

**Comment by cmcconnell1** [comment]
+1

**Comment by addisonbair** [comment]
+1

**Comment by mitchellh** [comment]
This should be fixed now (in 0.8 series)
